forui/lib/src/widgets/button/button.dart

```diff
@@ -262,6 +274,82 @@ class FButtonCustomStyle extends FButtonStyle with Diagnosticable {
   int get hashCode => enabledBoxDecoration.hashCode ^ disabledBoxDecoration.hashCode ^ content.hashCode ^ icon.hashCode;
 }
 
+/// [FButton] style for icon-button.
+class FButtonIconStyle extends FButtonStyle with Diagnosticable {
+  /// The box decoration for an enabled button.
+  final BoxDecoration enabledBoxDecoration;
+
+  /// The box decoration for a disabled button.
+  final BoxDecoration disabledBoxDecoration;
+
+  /// The content's style.
+  final FButtonContentStyle content;
+
+  /// The icon's style.
+  final FButtonIconStyle icon;
+
+  /// Creates a [FButtonIconStyle].
+  FButtonIconStyle({
+    required this.enabledBoxDecoration,
+    required this.disabledBoxDecoration,
+    required this.content,
+    required this.icon,
+  });
+
+  /// Returns a copy of this [FButtonIconStyle] with the given properties replaced.
+  ///
+  /// ```dart
+  /// final style = FButtonIconStyle(
+  ///   enabledBoxDecoration: ...,
+  ///   disabledBoxDecoration: ...,
+  ///   // other properties omitted for brevity
+  /// );
+  ///
+  /// final copy = style.copyWith(
+  ///   disabledBoxDecoration: ...,
+  /// );
+  ///
+  /// print(copy.background); // Colors.blue
+  /// print(copy.border); // Colors.black
+  /// ```
+  @useResult
+  FButtonIconStyle copyWith({
+    BoxDecoration? enabledBoxDecoration,
+    BoxDecoration? disabledBoxDecoration,
+    FButtonContentStyle? content,
+    FButtonIconStyle? icon,
+  }) =>
+      FButtonIconStyle(
+        enabledBoxDecoration: enabledBoxDecoration ?? this.enabledBoxDecoration,
+        disabledBoxDecoration: disabledBoxDecoration ?? this.disabledBoxDecoration,
+        content: content ?? this.content,
+        icon: icon ?? this.icon,
+      );
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties
+      ..add(DiagnosticsProperty('enabledBoxDecoration', enabledBoxDecoration))
+      ..add(DiagnosticsProperty('disabledBoxDecoration', disabledBoxDecoration))
+      ..add(DiagnosticsProperty('content', content))
+      ..add(DiagnosticsProperty('icon', icon));
+  }
+
+  @override
+  bool operator ==(Object other) =>
+      identical(this, other) ||
+      other is FButtonCustomStyle &&
+          runtimeType == other.runtimeType &&
+          enabledBoxDecoration == other.enabledBoxDecoration &&
+          disabledBoxDecoration == other.disabledBoxDecoration &&
+          content == other.content &&
+          icon == other.icon;
+
+  @override
+  int get hashCode => enabledBoxDecoration.hashCode ^ disabledBoxDecoration.hashCode ^ content.hashCode ^ icon.hashCode;
+}
+
```

Comment:
I think there's a slight misunderstanding, `FButtonIconContentStyle` should be a field in `FButtonCustomStyle`, i.e. `iconContent`.

Reply:
So there is a `FButtonIconContentStyle` which just contains a padding property for the icon-content. The `FButtonIconContentStyle` is also a property in `FButtonCustomStyle` named as `iconContent` which can be given by the user. This is then retrieved in `_FButtonIconContent` from the `FButton._of(...)` method and used. Is this right?

Reply:
> So there is a `FButtonIconContentStyle` which just contains a padding property for the icon-content. The `FButtonIconContentStyle` is also a property in `FButtonCustomStyle` named as `iconContent` which can be given by the user. This is then retrieved in `_FButtonIconContent` from the `FButton._of(...)` method and used. Is this right?

Yup sounds about right!